auto_nag/tests/test_user_activity.py

```diff
@@ -0,0 +1,98 @@
+from auto_nag.people import People
+from auto_nag.user_activity import UserActivity, UserStatus
+
+
+class UserActivityTest(MockTestCase):
+    """Test the UserActivity class"""
+
+    mock_urls = [BugzillaUser.URL]
+
+    active_users = {"smujahid@mozilla.com", "mcastelluccio@mozilla.com"}
+
+    disabled_users = {
+        "bdahl@mozilla.com",
+        "u709243@disabled.tld",
+        "balbeza@mozilla.com",
+    }
+
+    employees = {"abc@mozilla.com", "def@mozilla.com"}
+
+    people = People([{"mail": email} for email in employees])
+
+    @responses.activate
+    def test_check_users(self):
+        """Test the check_users method"""
+
+        user_activity = UserActivity(people=self.people)
+        inactive_users = user_activity.check_users(
+            self.active_users | self.disabled_users | self.employees
+        )
+
+        self.assertEqual(inactive_users.keys(), self.disabled_users)
+
+    @responses.activate
+    def test_check_users_keep_active(self):
+        """Test the check_users method with keep_active set to True"""
+
+        user_activity = UserActivity(people=self.people)
+        users_info = user_activity.check_users(
+            self.active_users | self.disabled_users | self.employees, keep_active=True
+        )
+
+        for email in self.active_users:
+            self.assertIn(email, users_info)
+
+            user = users_info[email]
+            self.assertNotIn("is_employee", user)
+            self.assertEqual(user["status"], UserStatus.ACTIVE)
+
+        for email in self.employees:
+            self.assertIn(email, users_info)
+
+            user = users_info[email]
+            self.assertEqual(user["is_employee"], True)
+            self.assertEqual(user["status"], UserStatus.ACTIVE)
+
+        for email in self.disabled_users:
+            self.assertIn(email, users_info)
+
+            user = users_info[email]
+            self.assertNotIn("is_employee", user)
+            self.assertEqual(user["status"], UserStatus.DISABLED)
+
+    @responses.activate
+    def test_get_bz_users_with_status(self):
+        """Test the get_bz_users_with_status method"""
+
+        user_activity = UserActivity(people=self.people, include_fields=["nick", "id"])
+        users_info = user_activity.get_bz_users_with_status(
+            list(self.active_users | self.disabled_users)
+        )
+
+        for email in self.active_users:
+            self.assertIn(email, users_info)
+
+            user = users_info[email]
+            self.assertIn("nick", user)
+            self.assertIn("id", user)
+            self.assertEqual(user["status"], UserStatus.ACTIVE)
+
+        for email in self.disabled_users:
+            self.assertIn(email, users_info)
+
+            user = users_info[email]
+            self.assertIn("nick", user)
+            self.assertIn("id", user)
+            self.assertEqual(user["status"], UserStatus.DISABLED)
+
+        user = users_info["bdahl@mozilla.com"]
+        user["id"] = 425126
```

Comment:
Are this and the next line needed?

Reply:
Sorry, it is supposed to be an asserion. Fixed in 8cb5c80c6fc7ff434bfa8f8c3bb0735500d76550.